### PR TITLE
Utility libxsmm Python extension

### PR DESCRIPTION
**Pull request by adam-smnk**
Adds a python wrapper for a parallelized in-place copy function using libxsmm and OpenMP.
It is intended to be used for efficient tensor padding implementation.

The libxsmm path have to be specified through env variables:
  - XSMM_ROOT_DIR - path to libxsmm root dir with headers
  - XSMM_LIB_DIR - path to libxsmm.so location

libxsmm .so also has to be available during runtime execution e.g., exposed through LD_LIBRARY_PATH.
The XSMM python module can be built and installed using command:
  pip install -e ./third_party/cpu/python/

### CODE REVIEW

**Review by rolfmorel — COMMENTED** [review]
Nit: `third_party/cpu/python/copy.cpp`, I have a feeling there might be a more appropriate name.

**Review by rolfmorel — APPROVED** [review]
LGTM.

File: third_party/cpu/python/xsmm_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#include <torch/extension.h>
+
+#include "libxsmm.h"
+#include <omp.h>
+
+#include <cstring>
+#include <vector>
+
+bool fastCopy2D(torch::Tensor &input, torch::Tensor &output) {
```
Comment (rolfmorel) [inline third_party/cpu/python/xsmm_utils.cpp:9]: I think this function is unused. I think.

Comment (adam-smnk) [reply third_party/cpu/python/xsmm_utils.cpp:9]: True, pad is also "safer" to some extent. I'll clean it up.

**Pull request merged**